solarforecastarbiter/reference_forecasts/forecast.py

```diff
@@ -0,0 +1,205 @@
+"""
+Functions for forecasting.
+"""
+
+from solarforecastarbiter import pvmodel
+
+# some functions originally implemented in pvlib-python.
+# find pvlib-python license in licenses directory.
+
+
+def cloud_cover_to_ghi_linear(cloud_cover, ghi_clear, offset=35):
+    """
+    Convert cloud cover to GHI using a linear relationship.
+    0% cloud cover returns ghi_clear.
+    100% cloud cover returns offset*ghi_clear.
+
+    Parameters
+    ----------
+    cloud_cover: numeric
+        Cloud cover in %.
+    ghi_clear: numeric
+        GHI under clear sky conditions.
+    offset: numeric, default 35
+        Determines the minimum GHI.
+
+    Returns
+    -------
+    ghi: numeric
+        Estimated GHI.
+
+    References
+    ----------
+    Larson et. al. "Day-ahead forecasting of solar power output from
+    photovoltaic plants in the American Southwest" Renewable Energy
+    91, 11-20 (2016).
+    """
+
+    offset = offset / 100.
+    cloud_cover = cloud_cover / 100.
+    ghi = (offset + (1 - offset) * (1 - cloud_cover)) * ghi_clear
+    return ghi
+
+
+def cloud_cover_to_irradiance_clearsky_scaling(cloud_cover, ghi_clear,
+                                               solar_zenith):
+    """
+    Estimates irradiance from cloud cover in the following steps:
+
+    1. Estimate cloudy sky GHI using a function of cloud_cover and
+       ghi_clear: :py:func:`cloud_cover_to_ghi_linear`
+    2. Estimate cloudy sky DNI and DHI using the Erbs model.
+
+    Parameters
+    ----------
+    site : datamodel.Site
+    cloud_cover : Series
+        Cloud cover in %.
+
+    Returns
+    -------
+    ghi : pd.Series, dni : pd.Series, dhi : pd.Series
+    """
+    ghi = cloud_cover_to_ghi_linear(cloud_cover, ghi_clear)
+    dni, dhi = pvmodel.complete_irradiance_components(ghi, solar_zenith)
+    return ghi, dni, dhi
+
+
+def cloud_cover_to_irradiance_clearsky_scaling_solpos(
```

Comment:
Seems that this function is a more logical entry point than the preceding two.  Perhaps the name could be shorted to `cloud_cover_to_irradiance` and `ghi_clear=None` added to the arguments.  We really should lay out the workflow, I think that would help.

Reply:
I'm ok with `cloud_cover_to_irradiance` here. I'm less sure about the conditional because of the large difference in required input data if `ghi_clear` is supplied or not.